File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestOptLongValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.values.OptLongInValue.OptLongValue;
import com.dslplatform.patterns.ServiceLocator;
import org.junit.*;
import static org.junit.Assert.*;

public class TestOptLongValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
//        locator = Bootstrap.init(TestOptLongValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
//        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
//        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertEquals(
                Long.class,
                OptLongValue.class.getDeclaredField("optLong").getGenericType());
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertEquals(
                Long.class,
                OptLongValue.class.getMethod("getOptLong").getGenericReturnType());
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertEquals(
                OptLongValue.class,
                OptLongValue.class.getMethod("setOptLong", Long.class).getReturnType());
    }

    @Test
    public void testDefaultPropertyValue() {
        assertNull(new OptLongValue().getOptLong());
    }
}

File: workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestOptLongValue.scala
package com.dslplatform.ocd.values

import OptLongInValue.OptLongValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestOptLongValue extends SpecExtensions {
  def `Test property field type: Option[Long]`() =
    checkType(
      typeOf[Option[Long]]
    , typeOf[OptLongValue].member("optLong": TermName).asMethod.returnType
    )

  def `Test property default value: None`() {
    assert(OptLongValue().optLong === None)
  }
}
